File: backend/models/skill.go
package models

import (
	"time"
)

type Skill struct {
	ID          uint           `gorm:"primaryKey" json:"id"`
	CategoryID  uint           `gorm:"index" json:"category_id"`
	Category    *SkillCategory `gorm:"foreignKey:CategoryID" json:"category,omitempty"`
	Name        string         `gorm:"size:100;not null" json:"name"`
	Proficiency int            `gorm:"default:80" json:"proficiency"` // 1-100
	Icon        string         `gorm:"size:100" json:"icon"`
	Order       int            `gorm:"default:0" json:"order"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
}

File: frontend/types/skill.ts
export interface Skill {
  id: number;
  category_id: number;
  name: string;
  proficiency: number;
  level?: string;
  icon?: string;
  order: number;
}

export interface SkillCategory {
  id: number;
  name: string;
  slug: string;
  description?: string;
  order: number;
  skills?: Skill[];
}